modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/Swift5ClientCodegen.java

```diff
@@ -1091,6 +1110,13 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
             // which provide Objective-C compatibility.
             property.vendorExtensions.put("x-swift-optional-scalar", true);
         }
+
+        if (property.isEnum) {
+            // A non-frozen (with the "unknownDefault" case) enum is only possible with String or Int raw types.
+            if (property.dataType.equals("String") || property.dataType.equals("Int")) {
```

Comment:
Since now we are going to assign values to the enums, can we remove this?

Reply:
Maybe. Let me push the next update. I think we'll want to know the enum's type in order to assign a default value that is correct.

Reply:
Is there any enum type that we don't support?
I think there is an easier way to do this, by removing this block of code and do this type check in the mustache files.
I think there are checks like isString, isInt, isDouble, isNumber, but I need to check tomorrow, because I'm not 100% sure.

Reply:
@4brunu You are correct. I had tried this earlier, but was using the mustache clauses incorrectly. I made the change to remove the custom vendor extensions and use the built-in type checks. I think we should be good now